### ISSUE TITLE
Proper window order for "Show Windows" (Expose) and "Show Workspaces"

### ISSUE THREAD

**Issue by ehansin** [linuxmint/Cinnamon#4323]
I waited to post this after the 2.6 release just in case it was fixed (it is not.)

If I have say a browser open on the left side of my desktop and a text editor on the right side (for example), when I trigger either "Show Windows" (Expose) or "Show Workspaces", the composite view of the desktop sometimes swaps the windows around, right now left and left now right. This then returns to original order/placement upon returning to normal desktop view.

For "Show Workspaces" view, if the workspace does not have focus, order/arrangement is correct, but as soon as I give focus to a workspace, sometimes it will swap the windows/apps placement again. And then back again upon losing focus, or as stated above, upon return to a single desktop view.

I am sure I am not the only one that finds this a bit annoying. Btw, just constructive criticism for a DE I do very much like a lot and want to see keep getting better. If I missed something in regards to this (settings, etc.) let me know, and sorry if this issue has been mentioned before; I'm a little new to GitHub.

Thanks!
Eric

**Comment by tp0** [comment]
At least you're not alone, I've always found this annoying too (also in other DEs, at least Unity & Compiz Spread comes to mind). It would be great if the position of the windows would stay the same at least for as long as no new windows are opened on the workspace.

This is further made confusing by not having the windows retain their real relative size, but that's another bug: #3336

**Comment by tp0** [comment]
I think you're using wrong terms: "Show Windows" is Scale and "Show Workspaces" is Expo. I don't quite understand the latter issue, do you mean in the Expo view when moving mouse over the workspace but not clicking the workspace?

I've always found this annoying too (also in other DEs, at least Unity & Compiz's Scale and Spread comes to mind). It would be great if the position of the windows would stay the same at least for as long as no new windows are opened on the workspace. I think this is a primary reason why I generally don't use Scale a lot... [I have no idea how/if the order technicallly relates to e.g. alt-tab switcher order]

Scale is further made more confusing by not having the windows retain their real relative size, but that's another bug: #3336

**Comment by tp0** [comment]
Dupe? #3906

**Comment by ehansin** [comment]
Okay, I used the wrong terms because I was thinking of how OS X named things, using Exposé for windows and Spaces for workspaces. So I see now that, roughly:

"Show all windows" (in Hot Corners) = "Exposé" (OS X) = "Scale" (Cinnamon applet)
"Show all workspaces" (in Hot Corners) = "Spaces" (OS X) = "Expo" (Cinnamon applet)

I updated the title to reflect proper terminology in Cinnamon.

Regardless, in Scale or in Expo (for workspace with focus), application windows get swapped left to right. So, left to right, A and B being my two applications open, each taking half the horizontal width of the screen:

In normal desktop "working" view:
A - B

But in Scale or Expo (for workspace with focus), often these get swapped to:
B - A

Then once I return out of Scale or Expo, back to normal desktop "working" view:
A - B

Yes, this looks possibly a duplicate of #3906, though it doesn't quite describe it as I do here. I have 16:9 monitor, and often have two apps open side by side. They should retain the same relative positioning in Scale or Expo, at least it makes sense to me, regardless of what application window last had focus.

Anyway, that is my take. Doesn't seem like #3906 has received much attention, but to me it also isn't exactly clear unless I am missing something. There is no "front" in what I am talking about, just left and right (no X-Y-Z axes, just X-Y.)

Eric
